src/crawler.ts

```diff
@@ -134,22 +134,23 @@ export abstract class Crawler {
 
     async waitForSelector(selector: string): Promise<ElementHandle | null> {
         logger.debug(`waitForSelector: ${selector}`);
-        await this.page.waitForTimeout(1000);
 
         return await this.page.waitForSelector(selector, { timeout: 2000 });
     }
 
     async waitAndClick(selector: string): Promise<void> {
         logger.debug(`waitAndClick: ${selector}`);
-        await this.page.waitForTimeout(1000);
 
         await this.page.waitForSelector(selector, { timeout: 2000 });
-        await this.page.evaluate(s => document.querySelector(s).click(), selector);
+        
+        await this.page.evaluate(s =>  {
```

Comment:
got an error here when building. I think you need to define the type string
```suggestion
        await this.page.evaluate((s: string) =>  {
```